src/meta-srv/src/service/store/etcd.rs

```diff
@@ -226,6 +207,72 @@ impl KvStore for EtcdStore {
             prev_kvs,
         })
     }
+
+    async fn move_value(&self, req: MoveValueRequest) -> Result<MoveValueResponse> {
+        let MoveValue {
+            cluster_id,
+            from_key,
+            to_key,
+            delete_options,
+        } = req.try_into()?;
+
+        let mut client = self.client.kv_client();
+
+        let kv = 'success: loop {
+            let from_key = from_key.as_slice();
+            let to_key = to_key.as_slice();
+
+            let res = client
+                .get(from_key, None)
+                .await
+                .context(error::EtcdFailedSnafu)?;
+
+            let txn = match res.kvs().first() {
+                None => {
+                    // get `to_key` if `from_key` absent
+                    // revision 0 means key was not exist
+                    let compare = Compare::create_revision(from_key, CompareOp::Equal, 0);
+                    let get = TxnOp::get(to_key, None);
+                    Txn::new().when(vec![compare]).and_then(vec![get])
+                }
+                Some(kv) => {
+                    // compare `from_key` and move to `to_key`
+                    let value = kv.value();
+                    let compare = Compare::value(from_key, CompareOp::Equal, value);
+                    let delete = TxnOp::delete(from_key, delete_options.clone());
+                    let put = TxnOp::put(to_key, value, None);
+                    Txn::new().when(vec![compare]).and_then(vec![delete, put])
+                }
+            };
+
+            let txn_res = client.txn(txn).await.context(error::EtcdFailedSnafu)?;
+
+            if !txn_res.succeeded() {
+                warn!(
+                    "Failed to atomically move {:?} to {:?}, try again...",
+                    String::from_utf8_lossy(from_key),
+                    String::from_utf8_lossy(to_key)
+                );
+                continue;
```

Comment:
Do we need to restrict the max retry times?

Reply:
Set a fixed number of attempts, maybe in the future we can let the users control it.